### DISCUSSION TITLE
Add to the server library LinuxGSM, a fork of Paper - Pufferfish.

### DISCUSSION THREAD

**Discussion by d3adwolf** [GameServerManagers/LinuxGSM#4039] [Ideas]
Greetings all, I suggest to add an excellent fork of the Paper kernel - Pufferfish, it is a highly optimized version of the usual Paper kernel aimed at maximum performance.

I have already talked to the developers and they will be quite happy to cooperate to add it, I can add that Pufferfish is already running in my LinuxGSM, by adding the line `executable="./pufferfish.jar"` in instance config and the jar file of the new kernel. 

As for functionality at the moment only update function doesn't work, so it will be necessary to implement only it, but guys have GitHub, so you can look in advance what and how it is set up there: https://github.com/pufferfish-gg/Pufferfish. To contact the developers you can use their Discord server: https://discord.gg/reZw4vQV9H.

If you liked my idea, you can support it with an arrow to raise it in the top list and write a comment on your experience with this kernel!

If you are interested, you can also read about the idea (#4040) of adding a new performance kernel proxy, Waterfall, to the LinuxGSM library. Thanks for your time and feedback!

**Comment by h3o66** [comment]
In general a integration is possible, but the following is needed:
- api to get the latest version ( the one on the website of from their CI looks like that can be used )
- enough interest from people to use lgsm with pufferfish
